File: Python/1290. Convert Binary Number in a Linked List to Integer.py
# O(N), O(1)
def getDecimalValue(self, head: ListNode) -> int:
        # num = head.val
        # while head.next:
        #     num = num * 2 + head.next.val
        #     head = head.next
        # return num 

        num = head.val
        while head.next:
            num = num << 1 | head.next.val
            head = head.next
        return num 

File: Golang/1290. Convert Binary Number in a Linked List to Integer.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
// O(N), O(1)
func getDecimalValue(head *ListNode) int {
	var res int
	for head != nil {
		res = (res << 1) | head.Val
		head = head.Next
	}
	return res
}

func getDecimalValue(head *ListNode) int {
	var res int
	for head != nil {
		res <<= 1
		res |= head.Val
		head = head.Next
	}
	return res
}

func getDecimalValue(head *ListNode) int {
	res := head.Val
	for head.Next != nil {
		res = res*2 + head.Next.Val
		head = head.Next
	}
	return res
}
